### ISSUE TITLE
How to Train YOLOv8 after installing via git clone route

### ISSUE THREAD

**Issue by AT9991** [ultralytics/ultralytics#4570]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have insalled YOLOv8 using git clone route on my colab notebook. After installing it when I want to train the model for detection using !yolo detect train model=yolov8n.yaml data=coco.yaml epoch=100. 

I get the following error:
/bin/bash: line 1: yolo: command not found

Can someone please help me how to train the model provided I have installed YOLOv8 using !git clone. Thank you!

I have attached the screenshot for your reference. 

<img width="1355" alt="Screenshot 2023-08-25 at 11 24 08" src="https://github.com/ultralytics/ultralytics/assets/64208663/d477ae18-2cfd-4fcb-937e-3c86f9357f7f">

**Comment by sOR-o** [comment]
I would suggest you to go through the official documentation https://docs.ultralytics.com/ of the yolov8 model their every step has been mentioned clearly. but in case if you are facing difficulty in understanding that so follow the following steps :

I would personally suggest to use python as interface rather than using CLI

1.  `!pip install ultralytics`
2.  Create a new YOLO model from scratch : `model = YOLO('yolov8n.yaml')`
      Load a pretrained YOLO model (recommended for training) : `model = YOLO('yolov8n.pt')`
3.  Prepare the dataset you wanna train on. (including the image and annotated txt file for every picture.)
4. Train the model using the 'coco128.yaml' dataset for required number of epochs(say x) : 
     `results = model.train(data='dataset.yaml', epochs=x)`
    Evaluate the model's performance on the validation set : `results = model.val()`
5. Perform object detection on an image using the model : `results = model('imagePath or url')`

i hope it will help you , if you have any further query feel free to drop it.

**Comment by AT9991** [comment]
Hi @sOR-o , Thanks for your comment. I wnat to install it via !git clone as I want to modify the network structure. In case of YOLOv5, I usually clone the whole repository and then change the necessary files and then train it using !python train.py. But in case of YOLOv8, if I install using pip, I am unable to access the required files. Is there a way, I install YOLOv8 using pip and also be able to acess the files? Thank you

**Comment by sOR-o** [comment]
@AT9991 you can use the `-e` flag to install it in `"editable"` mode. This creates a symbolic link to the package source code, allowing you to make modifications that are immediately reflected without reinstalling.

Clone the YOLOv8 repository : `git clone https://github.com/yolov8-repo.git`
Navigate to the repository directory : `cd yolov8-repo`
Install the package in editable mode : `pip install -e .`

Now you can do the modifications  in the cloned repository And your changes will be reflected when you use the package.

**Comment by glenn-jocher** [comment]
@AT9991, thank you for your beneficial contributions. A more direct way to edit YOLOv8 files that you've cloned is to employ Jupyter cells in Google Colab. Following your repository clone, you can make use of the command `%cd yolov8` to shift the directory to the cloned repository. This change will enable direct file access from the notebook. 

Remember to restart the Colab runtime or refresh any imported modules after conducting any modifications, ensuring your changes are loaded correctly. Modifications can be confirmed by running `!cat` to the specific file from the workbook for verification. 

I hope these insights provide some guidance. Please feel free to seek additional clarification.

**Comment by Jayasrikotipalli** [comment]
git clone https://github.com/yolov8-repo.git  i am trying to clone this particular repo but it shows error like fatal: repository 'https://github.com/yolov8-repo.git/' not found

**Comment by glenn-jocher** [comment]
@Jayasrikotipalli hi there! It looks like the repository URL might be incorrect or the repository is private. Could you double-check the URL? If it's a private repo, ensure you have access. For public repos, the correct format usually looks like this: `git clone https://github.com/ultralytics/yolov8.git`. Let me know if this helps! 😊

**Comment by kanis777** [comment]
git clone https://github.com/ultralytics/yolov8.git "repo not found error ". So, can you tell if the git has been moved

**Comment by glenn-jocher** [comment]
Hello! It seems like there might be an issue with the repository URL. Please ensure that the URL is correct and that the repository is still available at that location. If it's a private repository, you'll need access permissions to clone it. If the issue persists, the repository might have been moved or renamed. Checking with the repository owner or the Ultralytics official page for the updated URL could be helpful. Let me know if you need further assistance! 😊

**Comment by TimotheeWright** [comment]
> Checking with the repository owner or the Ultralytics official page for the updated URL could be helpful
Aren't you the owner ?

Anyway same question for me I wanted to know if I should clone all [https://github.com/ultralytics/ultralytics.git](https://github.com/ultralytics/ultralytics.git) or if I could clone something more specific to yolov8  ? I believe in the ultralytics repo there is more than just yolov8 related files

**Comment by glenn-jocher** [comment]
Hello! Yes, you're right—I'm here to help! 😄 For YOLOv8 specifically, you should clone the main Ultralytics repository as it includes YOLOv8 among other models. Currently, all Ultralytics models are managed within the same repository, so there isn't a separate repository exclusively for YOLOv8. Cloning the main repository will give you access to all the models and tools you need. If you have any more questions or need further assistance, feel free to ask. Happy coding! 🚀
